src/SoundContext.cpp

```diff
@@ -64,17 +65,18 @@ void SoundContext::init() {
 }
 
 void SoundContext::addSoundToLoad(SoundInfo* info) {
-    soundsToLoad_->push_back(info);
+    (*soundsToLoad_)[info->id_] = info;
 }
 
-FMOD::Sound* SoundContext::getSound(const std::string& id) {
-    return sounds_[id];
-}
-
-Channel* SoundContext::playSound(FMOD::Sound* sound) const {
+Channel* SoundContext::playSound(const std::string& id) const {
     FMOD::Channel* channel;
     try {
-        auto result = system_->playSound(sound, 0, false, &channel);
+        auto result =
+            system_->playSound(getInstance()->getSound(id), 0, true, &channel);
+        ERRCHECK(result);
+        result = channel->setVolume((*soundsToLoad_)[id]->volume_);
+        ERRCHECK(result);
```

Comment:
You should declare `(*soundsToLoad_)[id]->volume_` as a variable, then check if it's **not** near 1.0 (`abs(1.0 - volume) > 0.01`, or just check `volume != 1.0` if you don't care about security), in which case you should set up the volume.

Why is that? Because changing the volume of audios have a very large performance impact, if an audio is always supposed to play at `1.0` (unmodified/default value), then it should not have its volume set.

As always, please correct me if I'm wrong 👍🏼

Reply:
changed